cellrank/kernels/_velocity_kernel.py

```diff
@@ -63,16 +65,34 @@ def _read_from_adata(
         self,
         xkey: Optional[str] = "Ms",
         vkey: Optional[str] = "velocity",
+        attr: Optional[Literal["layers", "obsm"]] = "layers",
         gene_subset: Optional[Union[str, Sequence[str]]] = None,
         **kwargs: Any,
     ) -> None:
         super()._read_from_adata(**kwargs)
 
-        if gene_subset is None and f"{vkey}_genes" in self.adata.var:
+        if attr not in ["layers", "obsm"]:
```

Comment:
I think the condition can be simplified to and remove the warning + mention in the docstring of `gene_subset` it only available for `attr = 'layers'`.
```python
if attr == "layers" and gene_subset is None and f"{vkey}_genes" in self.adata.var:
   <as before>
self._x_data = ...
```

Reply:
I will need an `elif` or `else` in any case in order to set `gene_subset = None` when reading from obsm - and then I think it would make sense to keep the warning. Or is there any negative side effect that I am missing?